packages/std/src/ibc.rs

```diff
@@ -213,37 +211,136 @@ impl IbcAcknowledgement {
 
 /// The message that is passed into `ibc_channel_open`
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-pub struct IbcChannelOpenMsg {
-    pub channel: IbcChannel,
+pub enum IbcChannelOpenMsg {
+    OpenInit {
+        channel: IbcChannel,
+    },
+    OpenTry {
+        channel: IbcChannel,
+        counterparty_version: String,
+    },
 }
 
 impl IbcChannelOpenMsg {
-    pub fn new(channel: IbcChannel) -> Self {
-        Self { channel }
+    pub fn new_init(channel: IbcChannel) -> Self {
+        Self::OpenInit { channel }
+    }
+
+    pub fn new_try(channel: IbcChannel, counterparty_version: impl Into<String>) -> Self {
+        Self::OpenTry {
+            channel,
+            counterparty_version: counterparty_version.into(),
+        }
+    }
+
+    pub fn channel(&self) -> &IbcChannel {
+        match self {
+            Self::OpenInit { channel } => channel,
+            Self::OpenTry { channel, .. } => channel,
+        }
+    }
+
+    pub fn counterparty_version(&self) -> Option<&str> {
+        match self {
+            Self::OpenTry {
+                counterparty_version,
+                ..
+            } => Some(counterparty_version),
+            _ => None,
+        }
+    }
+}
+
+impl From<IbcChannelOpenMsg> for IbcChannel {
+    fn from(msg: IbcChannelOpenMsg) -> IbcChannel {
+        match msg {
+            IbcChannelOpenMsg::OpenInit { channel } => channel,
+            IbcChannelOpenMsg::OpenTry { channel, .. } => channel,
+        }
     }
 }
 
 /// The message that is passed into `ibc_channel_connect`
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-pub struct IbcChannelConnectMsg {
-    pub channel: IbcChannel,
+pub enum IbcChannelConnectMsg {
+    ConnectAck {
+        channel: IbcChannel,
+        counterparty_version: String,
+    },
+    ConnectConfirm {
```

Comment:
"connect" is a term invented by Ethan/CosmWasm that is not from the IBC spec. The original terms are ChanOpenInit, OpenTry, ChanOpenAck, ChanOpenConfirm, ChanCloseInit, ChanCloseConfirm (https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#channel-lifecycle-management). I'd stick whith those, but just drop the "chan" part.

Reply:
Ah, maybe that's why I originally did it like that. Reverting...

Reply:
Could you add a doc comment to all enum cases like this (or similar)?

```
/// The ChanOpenConfirm step from https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics#channel-lifecycle-management
OpenConfirm {
```